evennia/contrib/fieldfill.py

```diff
@@ -0,0 +1,561 @@
+        object dbrefs).
+"""
+
+from evennia.utils import evmenu, evtable, delay, list_to_string
+from evennia import Command
+from evennia.server.sessionhandler import SESSIONS
+
+class FieldEvMenu(evmenu.EvMenu):
+    """
+    Custom EvMenu type with its own node formatter - removes extraneous lines
+    """
+    
+    def node_formatter(self, nodetext, optionstext):
+        """
+        Formats the entirety of the node.
+
+        Args:
+            nodetext (str): The node text as returned by `self.nodetext_formatter`.
+            optionstext (str): The options display as returned by `self.options_formatter`.
+            caller (Object, Account or None, optional): The caller of the node.
+
+        Returns:
+            node (str): The formatted node to display.
+
+        """
+        # Only return node text, no options or separators
+        return nodetext
+    
+
+def init_fill_field(formtemplate, caller, callback, pretext="", posttext="",
+                    submitcmd="submit", borderstyle="cells", helptext=None):
+    """
+    Initializes a menu presenting a player with a fillable form - once the form
+    is submitted, the data will be passed as a dictionary to your chosen
+    function.
+    
+    Args:
+        formtemplate (list of dicts): The template for the form's fields
+        caller (obj): Player who will be filling out the form
+        callback (callable): Function to pass the completed form's data to
+        
+    Options:
+        pretext (str): Text to put before the form in the menu
+        posttext (str): Text to put after the form in the menu
+        submitcmd (str): Command used to submit the form
+        borderstyle (str): Form's EvTable border style
+        helptext (str): Help text for the form menu (or default is provided)
+    """
+    # Initialize form data from the template
+    blank_formdata = form_template_to_dict(formtemplate)
+    
+    # Provide default help text if none given
+    if helptext == None:
+        helptext = ("Available commands:|/"
+                     "|w<field> = <new value>:|n Set given field to new value, replacing the old value|/"
+                     "|wclear <field>:|n Clear the value in the given field, making it blank|/"
+                     "|wlook|n: Show the form's current values|/"
+                     "|whelp|n: Display this help screen|/"
+                     "|wquit|n: Quit the form menu without submitting|/"
+                     "|w%s|n: Submit this form and quit the menu" % submitcmd)
+    
+    # Pass kwargs to store data needed in the menu
+    kwargs = {
+    "formdata":blank_formdata,
+    "formtemplate": formtemplate,
+    "callback": callback,
+    "pretext": pretext,
+    "posttext": posttext,
+    "submitcmd": submitcmd,
+    "borderstyle": borderstyle,
+    "helptext": helptext
+    }
+    
+    # Initialize menu of selections
+    FieldEvMenu(caller, "evennia.contrib.fieldfill", startnode="menunode_fieldfill", **kwargs)
```

Comment:
Not yet! I'll see about that when I go back to revisions. The only problem I foresee is having to specify db instead of ndb when storing and retrieving the menu data if the menu is specified as permanent.